<!doctype html>
<script
  src="https://www.datadoghq-browser-agent.com/datadog-rum-v4.js"
  type="text/javascript">
</script>

<script>
    window.DD_RUM && window.DD_RUM.init({
      clientToken: 'pub1689039fc57d50e8580aebfe8afd4580',
      applicationId: '971244fe-b5e5-46cf-a117-2332e27f76c8',
      site: 'datadoghq.com',
      service: 'bensunderland.github.io',
      env: 'dev',
      // Specify a version number to identify the deployed version of your application in Datadog 
      version: '1.0.0',
      sampleRate: 100,
      sessionReplaySampleRate: 100,
      trackInteractions: true,
      trackResources: true,
      trackLongTasks: true,
      defaultPrivacyLevel: 'allow',
    });

  function CustomException(message, metadata) {
    const error = new Error(message);
    error.metadata = metadata;
    DD_RUM.addError(error,  {
        source: 'source',
    });

    return error;
  }
  
  function onCheckoutButtonClick(cart) {
    window.DD_RUM && DD_RUM.addAction('checkout', {
        'value':  Math.floor(Math.random() * 100),
        'items': ['tomato', 'strawberries']
    });
    }
  
  function onErrorButtonClick() {
    CustomException("Error: My custom error", "a:b");
    }

  window.DD_RUM &&
  window.DD_RUM.startSessionReplayRecording();
</script>


<html>
<h1 >You did it!</h1>
<a href="https://app.datadoghq.com/rum/list">https://app.datadoghq.com/rum/list</a>

<input type="button" onclick="onCheckoutButtonClick();" value="Checkout Now" />

<input type="button" onclick="onErrorButtonClick();" value="Cause Error" />


</body>
</html>
